shared/domains/chelonia/chelonia.js

```diff
@@ -33,6 +39,78 @@ export type ChelActionParams = {
   server?: string; // TODO: implement!
   contractID: string;
   data: Object;
+  signingKeyId: string;
+  encryptionKeyId: ?string;
+  hooks?: {
+    prepublishContract?: (GIMessage) => void;
+    prepublish?: (GIMessage) => void;
+    postpublish?: (GIMessage) => void;
+  };
+  publishOptions?: { maxAttempts: number };
+}
+
+export type ChelKeyAddParams = {
+  contractName: string;
+  contractID: string;
+  data: GIOpKeyAdd;
+  signingKeyId: string;
+  hooks?: {
+    prepublishContract?: (GIMessage) => void;
+    prepublish?: (GIMessage) => void;
+    postpublish?: (GIMessage) => void;
+  };
+  publishOptions?: { maxAttempts: number };
+}
+
+export type ChelKeyDelParams = {
+  contractName: string;
+  contractID: string;
+  data: GIOpKeyDel;
+  signingKeyId: string;
+  hooks?: {
+    prepublishContract?: (GIMessage) => void;
+    prepublish?: (GIMessage) => void;
+    postpublish?: (GIMessage) => void;
+  };
+  publishOptions?: { maxAttempts: number };
+}
+
+export type ChelKeyShareParams = {
+  originatingContractID?: string;
+  originatingContractName?: string;
+  destinationContractID: string;
+  destinationContractName: string;
+  data: GIOpKeyShare;
+  signingKeyId: string;
+  hooks?: {
+    prepublishContract?: (GIMessage) => void;
+    prepublish?: (GIMessage) => void;
+    postpublish?: (GIMessage) => void;
+  };
+  publishOptions?: { maxAttempts: number };
+}
+
+export type ChelKeyRequestParams = {
+  originatingContractID: string;
+  originatingContractName: string;
+  contractName: string;
+  contractID: string;
+  signingKey: Key,
+  innerSigningKeyId: string;
+  encryptionKeyId: string;
+  hooks?: {
+    prepublishContract?: (GIMessage) => void;
+    prepublish?: (GIMessage) => void;
+    postpublish?: (GIMessage) => void;
+  };
+  publishOptions?: { maxAttempts: number };
+}
+
+export type ChelKeyRequestResponseParams = {
+  contractName: string;
+  contractID: string;
+  data: GIOpKeyRequestResponse;
+  signingKeyId: string;
```

Comment:
With so many types, maybe it makes sense to put them int their own individual `types.js` file?

Reply:
Yes, that sounds reasonable.